### PR TITLE
Frontend: Cleanup (1)

### PR DESCRIPTION
**Pull request by scouillard**
## Description

- Change Modal component fto be more general - last template was too restrictive.
- Change Avatar component to be more general - add URL as props, so Avatar can now be used anywhere.
- Add new SearchBar component in components/shared (has a search icon + custom x button to remove text)
- Clean-up the tree a bit.
-

### CODE REVIEW

File: app/javascript/components/shared/Modal.jsx [hunk 1]
```diff
@@ -3,34 +3,31 @@ import { Modal as BootstrapModal } from 'react-bootstrap';
...
 export default function Modal({
-  modalButton, title, body, footer,
+  modalButton, title, body,
 }) {
   const [show, setShow] = useState(false);
 
   const handleClose = () => setShow(false);
   const handleShow = () => setShow(true);
 
-  const Footer = React.cloneElement(footer, { handleClose });
```
Comment (farhatahmad) [inline app/javascript/components/shared/Modal.jsx:13]: What's your reasoning for removing the footer?

File: app/javascript/hooks/queries/users/useUsers.jsx [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { useQuery } from 'react-query';
+import axios from 'axios';
+
+export default function useUsers() {
+  return useQuery('getUsers', async () => axios.get('/api/v1/users.json', {
```
Comment (farhatahmad) [inline app/javascript/hooks/queries/users/useUsers.jsx:5]: What's this query used for?

**Comment by farhatahmad** [comment]
Make sure you run `eslint`

File: app/javascript/components/shared/Modal.jsx (hunk 1 above)
Comment (scouillard) [reply app/javascript/components/shared/Modal.jsx:13]: The modal template was too restrictive - it is not possible not make a one size fits all.
The footer doesn't bring any added value - it saves two or three CSS classes at most.
Removing the footer gives us more flexibility for the design.

File: app/javascript/hooks/queries/users/useUsers.jsx (hunk 2 above)
Comment (scouillard) [reply app/javascript/hooks/queries/users/useUsers.jsx:5]: It's in relation with the User#index that returns all Users.
https://github.com/bigbluebutton/greenlight/pull/3446#discussion_r873982483

**Comment by scouillard** [comment]
The eslint error is due to the SharedAccess component not being created yet. PR #3447

File: app/javascript/hooks/queries/users/useUsers.jsx (hunk 2 above)
Comment (farhatahmad) [reply app/javascript/hooks/queries/users/useUsers.jsx:5]: See my comment there

**Comment by farhatahmad** [comment]
> The eslint error is due to the SharedAccess component not being created yet. It is created in PR https://github.com/bigbluebutton/greenlight/pull/3447

PR's can't be merge if they're failing the pipeline. Choose one of the PRs and move the changes there

**Review by farhatahmad — APPROVED** [review]

**Pull request merged**